### PR TITLE
Parsed script files to json

### PR DESCRIPTION
**Pull request by goelakash**
@henrykironde @ethanwhite 
I modified the `compile.py` parser to save the `.script` files to `json` in a temp folder for review purposes.

Some ideas to follow `datapackage.json` standard might be-

1. 'ref' -> 'homepage'
2. 'urls' -> 'resources' (with name and url of tables as sub-keys)

### CODE REVIEW

**Comment by henrykironde** [comment]
@goelakash, great start. 
One of the reasons for the shift to json is to enhance readability. So something like this 
```json
   {
"citation": "Susana Rodriguez-Buritica, Helen Raichle, Robert H. Webb, Raymond M. Turner, and D. Lawrence Venable. 2013. One hundred and six years of population and community dynamics of Sonoran Desert Laboratory perennials. Ecology 94:976.",
  "urls": {
    "Stake_info": "http://esapubs.org/archive/ecol/E094/083/Stake_info.csv",
    "SMDensity": "http://esapubs.org/archive/ecol/E094/083/SMDensity.csv",
    "Plot_corners": "http://esapubs.org/archive/ecol/E094/083/Plot_corners.csv",
    "Count1906": "http://esapubs.org/archive/ecol/E094/083/Count1906.csv",
    "Photo_info": "http://esapubs.org/archive/ecol/E094/083/Photo_info.csv",
    "Seedling_counts": "http://esapubs.org/archive/ecol/E094/083/Seedling_counts.csv",
    "Plots": "http://esapubs.org/archive/ecol/E094/083/Plots.csv",
    "SMCover": "http://esapubs.org/archive/ecol/E094/083/SMCover.csv",
    "Species": "http://esapubs.org/archive/ecol/E094/083/Species.csv"
  },
  "shortname": "SonoranPerennials",
  "description": "The data set constitutes all information associated with the Spalding-Shreve permanent vegetation plots from 1906 through 2012, which is the longest-running plant monitoring program in the world."
}
```

you can use online editors to improve the structure.

**Comment by goelakash** [comment]
@henrykironde How does this look?

**Comment by ethanwhite** [comment]
@goelakash - Thanks for your work on this. It looks really good as a proof of concept and definitely demonstrates that you'd be a good fit for this project.

Having converted the existing `.script` files to JSON next steps would include:

1. Updating `compile.py` to allow it to read JSON, but probably also maintaining the current ability to read our YAML like format for backwards compatibility.
2. Updating associated documentation
3. Potentially improving how we designate cross-tabulated data and how to convert it from "wide" to "long". The current way of describing this is confusing and since we're changing the scripting system this is the perfect time to fix it. Also fixing some associated bugs along the way (e.g., #241).
4. Writing a tool to let scientists who aren't familiar with JSON easily create JSON files for new datasets

Of course these are the sorts of things you'd have the time to do over the summer if your proposal was funded, so probably the next step is to go ahead and draft a proposal that references this as the first step and lays out your plan for tackling the broader set of work.

I look forward to chatting more.

**Comment by goelakash** [comment]
@ethanwhite Thanks for your input :D

1. I guess I can add this function as `compile_json` in the `compile.py` file, instead of modifying the present one as doing that may introduce some bugs.

2. I will update the docs to point out the we are migrating to a JSON standard.

3. Will have to look into this. Can I see some examples for this?

4. I am not familiar how they are creating script files, so where should I look for that?

I will add your points in my current draft and post a link here.

**Comment by henrykironde** [comment]
@goelakash  
4. the best way to start with scripts is by looking at the 
[Writing and loading Scripts for the EcoData Retrieverdocs](http://retriever.readthedocs.org/en/latest/scripts.html)

**Comment by goelakash** [comment]
I saw an issue to migrate current test-suite from nose to py.test. I think I can work on that too.

**Comment by henrykironde** [comment]
@goelakash sounds like a plan.
some of the issues will have to be taken care of as we shift to Json. 
We shall go through the draft and give you feed back.

**Comment by goelakash** [comment]
@ethanwhite @henrykironde Have you guys gone over the draft? Anything I should edit?

**Comment by henrykironde** [comment]
@goelakash, we are in the process of revising the draft and I will meet with @ethanwhite and have the changes noted out for you by Monday.

**Comment by goelakash** [comment]
I saw the comments and edited the draft. I have a few doubts:
1. We should be aiming for backwards compatibility, and for that we would have to use `six` and `modernize`/`pasteurize` packages.
2. There is a new version of wxpython out (project Phoenix) that is backwards compatible is well. Should I add migrating to that as well as part of my proposal? ( I can start working on a fork next week).

**Comment by ethanwhite** [comment]
> If we are aiming for backwards compatibility, we would have to use six and modernize/pasteurize packages instead of 2to3.

Good point. I like the language you added instead.

> There is a new version of wxpython out (project Phoenix) that is backwards compatible as well. Should I add migrating to that as well as part of my proposal? ( I can start working on a fork next week).

What is Phoenix's status? The last time I looked it wasn't complete and hadn't moved in a while.

Also, we're actively considering removing the GUI (see #441). We've sunk a lot of hours into maintaining a cross-platform GUI over the years and my feeling at the moment is that most ecologists have moved over to scripting their analyses in Python or R making it more important to improve our R wrapper and add a proper Python interface.

**Comment by ethanwhite** [comment]
Oh, and I forgot to say that your current proposal is looking great. I made a round of fairly minor comments and writing suggestions that should improve things, but I think it's really close to being ready to submit.

**Comment by ethanwhite** [comment]
Returning to our discussion of this PR:

> I guess I can add this function as compile_json in the compile.py file, instead of modifying the present one as doing that may introduce some bugs.

My thinking as that we would really only use the code for converting the `.script` files to JSON once and therefore wouldn't need to include it in the package in the end, but it's fine to store it there for now. As we've discussed we will want functions and logic in `compile.py` to check for both JSON and .script files and handle each one appropriately. If you don't have a note about this approach to maintaining backward compatibility in the proposal that might be worth adding.  

> I am not familiar how they are creating script files, so where should I look for that?

Write now we are basically just documenting how to do it and then people are writing them by hand. This is fairly easy to do with the YAML like scripts, but moving to JSON will make this more difficult for the average scientist. So I'm envisioning something like the `retriever add` functionality I described in a note in your proposal.

**Comment by goelakash** [comment]
@ethanwhite @henrykironde Yes, thanks for the edits. 

Wrt the PR, I moved the function to separate file. We could just use it as a helper tool, or delete it if there aren't any expectations of a new .script file being created in the next few months.

**Comment by goelakash** [comment]
@henrykironde So if I remove the py file, this PR can be merged?

**Comment by henrykironde** [comment]
@goelakash we have been busy with some work and your proposals so we did't get time to check the PR.

**Comment by ethanwhite** [comment]
I think there's an open question about how to best proceed with this transition. My initial thought is that we would want to develop this all the way before merging into `master`. This could be done either in your fork and then come in in one big PR, or we can set up a development branch in the main repo and PR the pieces into that branch one at a time until it's finished and then merge the branch into master. The later approach is probably the easiest to handle in terms of collaboration and review, without risking breaking `master` (or ending up with lots of not yet actively used files lying around) before the entire transition is ready. This lets us merge and review PRs, and work together, without having a partially transitioned state from `.script` to `.json`.

I have pushed the following branch for this purpose:
https://github.com/weecology/retriever/tree/scripts-to-json

@goelakash - if you can submit a new PR from your fork to that branch then @henrykironde can do another round of review (I think we should at least add a docstring to the file) and then get this merged.

We should also set up an issue with all of the steps necessary to complete this transition to make it easy to keep track of what has been completed and what is left to be done.

**Comment by goelakash** [comment]
@ethanwhite @henrykironde So Google funds our project now! :smile:

**Comment by ethanwhite** [comment]
Congratulations @goelakash! We're super excited to have you on board this summer.

**Comment by goelakash** [comment]
@ethanwhite So w.r.t the community bonding period, I have a few questions:
1. Is there a slack or gitter channel that I can join?
2. I currently blog at [Wordpress](goelakash.wordpress.com) . Should I setup a new  blog? I was thinking of using Jekyll for Github pages.
Cheers!

**Comment by henrykironde** [comment]
@goelakash, We use slack and we shall definitely create our channel or otherwise.
As for the blog, you can use your current platform. But Since the organization that is responsible for this summer program utilizes GitHub a lot, It may be good to use GitHub. We are also planning to keep a tab on our Google summer coders, so we shall need a few lines of your intro and and a photo and we shall link that to your blog.

**Comment by ethanwhite** [comment]
@goelakash 
1. I just added you to my groups Slack #software channel. Both @henrykironde and I hang out there and it's integrated with GitHub and Travis.
2. It's totally fine to continue to work with your current blog. That's what I'd do since you already have one established. I use Wordpress for blogging myself: https://jabberwocky.weecology.org/

File: scripts/AmnioteLH.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
+{
+    "citation": "Myhrvold, N.P., Baldridge, E., Chan, B., Sivam, D., Freeman, D.L. and Ernest, S.M., 2015. An amniote life-history database to perform comparative analyses with birds, mammals, and reptiles:Ecological Archives E096-269. Ecology, 96(11), pp.3109-000.",
+    "description": "Compilation of life history traits for birds, mammals, and reptiles.",
+    "name": "Amniote life History database",
+    "ref": "http://esapubs.org/archive/ecol/E096/269",
+    "shortname": "AmnioteDB",
+    "tables": {
+        "main": {
+            "cleanup": "Cleanup(correct_invalid_value, nulls=[-999])",
```
Comment (goelakash) [inline scripts/AmnioteLH.json]: @henrykironde Is this a correct representation? The earlier representation was:
```
"cleanup": {
                "nulls": [
                    -999
                ]
            }
```

File: scripts/EA_adler2007.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+    "citation": "Peter B. Adler, William R. Tyburczy, and William K. Lauenroth. 2007. Long-term mapped quadrats from Kansas prairie:demographic information for herbaceaous plants. Ecology 88:2673.",
+    "description": "Demographic data for testing current theories in plant ecology and forecasting the effects of global change.",
+    "name": "Kansas plant quadrats (Ecological Archives 2007)",
+    "shortname": "Adler2007",
+    "tables": {
+        "monthly_ppt": {
+            "cleanup": "Cleanup(correct_invalid_value, nulls=['NA'])",
+            "replace_columns": "[('jan', 'january'), ('feb', 'february'), ('mar', 'march'), ('apr', 'april'), ('jun', 'june'), ('jul', 'july'), ('aug', 'august'), ('sep', 'september'), ('oct', 'october'), ('nov', 'november'), ('dec', 'december')]"
```
Comment (goelakash) [inline scripts/EA_adler2007.json]: @henrykironde Also here, the earlier representation was (ignore the `-` at the beginning):
```
"replace_columns": [
-                [
-                    "jan",
-                    "january"
-                ],
-                [
-                    "feb",
-                    "february"
-                ],
-                [
-                    "mar",
-                    "march"
-                ],
-                [
-                    "apr",
-                    "april"
-                ],
-                [
-                    "jun",
-                    "june"
-                ],
-                [
-                    "jul",
-                    "july"
-                ],
-                [
-                    "aug",
-                    "august"
-                ],
-                [
-                    "sep",
-                    "september"
-                ],
-                [
-                    "oct",
-                    "october"
-                ],
-                [
-                    "nov",
-                    "november"
-                ],
-                [
-                    "dec",
-                    "december"
-                ]
-            ]
```

File: scripts/AmnioteLH.json (hunk 1 above)
Comment (henrykironde) [inline scripts/AmnioteLH.json]: I think that is correct, I will look at it more later and let you know if changes are needed

File: scripts/EA_adler2007.json (hunk 2 above)
Comment (goelakash) [inline scripts/EA_adler2007.json]: @henrykironde Also, should it be 
`"[('jan', 'january'), ('feb', 'february')]"` 
OR
`["('jan', 'january')", "('feb', 'february')"]`
?

Comment (henrykironde) [inline scripts/EA_adler2007.json]: yah I am thinking` "[('jan', 'january'), ('feb', 'february')]",` I have not yet tested or looked at this,

I suggest that you test both and see what you get. Also a reminder, as you work on this part, try to keep the tests that you have used, we could include some of them as test cases later.

**Pull request merged**